app/controllers/notify_controller.rb

```diff
@@ -0,0 +1,62 @@
+class NotifyController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  def callback
+    return render_unauthorized if unauthorized?
+    return render_unprocessable_entity if reference_or_status_missing?
+
+    response = ProcessNotifyCallback.call(notify_reference: params['reference'], status: params['status'])
+
+    if response == :not_found
+      render_not_found
+    else
+      render json: nil, status: :ok
+    end
+  end
+
+private
+
+  def unauthorized?
+    authorization_token.nil? || authorization_token != ENV.fetch('GOVUK_NOTIFY_CALLBACK_API_KEY')
+  end
+
+  def authorization_token
+    request.headers['Authorization']&.gsub('Bearer ', '')
+  end
```

Comment:
Rails actually has a - well-hidden - feature to to this stuff, you can see it in action here:

https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/commit/36c71e86d37e4e4fbffee969ee839edc4645edf3